### PR TITLE
GitHub CI, README and BASH Improvements

### PR DESCRIPTION
**Pull request by jp1337**
GitHub CI tests the following aspects:
- Markdownlint for README
- Shellcheck for Shell Scripts
- Execute a test build on Ubuntu 20.04

Shell Improvements:
- Fix Shebang black line
- Variable Escaping
- Parameters with integer instead of strings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+It is possible to display the dependencies between the classes.
+For this purpose the script dev_graph.sh in the tools folder can be used.
+
+Requirements:
+
+- package **graphviz** on Debian
+
+Arguments:
+
+- -p _print_graph_ instead of creating an image, the structure is printed in JSON format
```
Comment (iTitus) [inline README.md]: this option does not require an argument, it is a flag. Can you make that clearer?
Also the format is called [DOT](https://en.wikipedia.org/wiki/DOT_%28graph_description_language%29).

File: README.md [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+- -l _include_libs_ when creating the graph the libraries are displayed in addition to the own classes
```
Comment (iTitus) [inline README.md]: This is also just a flag

File: README.md [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+
+## requirements
+
+The following requirements must be met to form the project.
+
+On Debian the following packages must be installed:
+
+- libglfw3-dev
+- libfreetype-dev
+- g++
```
Comment (iTitus) [inline README.md]: Can we instead use `build-essential` here? This should cover all bases and install gcc, make and the libraries
```suggestion
- build-essential
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Check out the Repository
+        uses: actions/checkout@master
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libglfw3-dev libfreetype-dev g++
```
Comment (iTitus) [inline .github/workflows/build.yml]:
```suggestion
          sudo apt-get install -y libglfw3-dev libfreetype-dev build-essential
```

**Review by iTitus — CHANGES_REQUESTED** [review]
Thanks a lot, but there a few nitpicks left :)

File: tools/dep_graph.sh [hunk 5]
```diff
@@ -37,14 +37,14 @@ fi
     {
       cd "$source_dir" || exit 1
       grep -rH --include="*.c" --include="*.h" "^#include \""
-      if test "x$include_libs" != "x"; then
+      if [ $include_libs -gt 0 ]; then
```
Comment (iTitus) [inline tools/dep_graph.sh]: There is some inconsistency here. I think you need to do this:
```suggestion
      if [ "$include_libs" -ne 0 ]; then
```

File: tools/dep_graph.sh [hunk 6]
```diff
@@ -37,14 +37,14 @@ fi
...
         grep -rH --include="*.c" --include="*.h" "^#include <"
       fi
     } | sed -n "s/\(.*\):#include [\"<]\(.*\)[\">]/\t\"\1\" -> \"\2\";/p"
   done
 
   echo "}"
-} | if test "x$print_graph" = "x"; then
+} | if [ "${print_graph}" -ge 0 ]; then
```
Comment (iTitus) [inline tools/dep_graph.sh]:
```suggestion
} | if [ "${print_graph}" -ne 0 ]; then
```

**Review by iTitus — CHANGES_REQUESTED** [review]

**Comment by jp1337** [comment]
Resolved the issues in latest commits 😄 
Found a typo and corrected it directly.

**Pull request merged**